### PR TITLE
[uss_qualifier/common_dict_eval] Move track to generic function

### PR DESCRIPTION
**Pull request by the-glu**
With tests.

Following @BenjaminPelletier remarks in https://github.com/interuss/monitoring/pull/940, this moves track direction checks to the _generic_evaluator, improving tests and edge case handling.

### CODE REVIEW

File: monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py [hunk 1]
```diff
@@ -456,51 +452,49 @@ def value_comparator(v1: Optional[float], v2: Optional[float]) -> bool:
...
+                )
+
+            return val
+
+        def value_comparator(v1: Optional[float], v2: Optional[float]) -> bool:
+
+            if v1 is None or v2 is None:
+                return False
+
+            return abs(v1 % 360 - v2 % 360) < constants.MinTrackDirectionResolution
```
Comment (mickmis) [inline monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py:484]: Not sure about that: why applying `% 360` on v1 and v2 and not just computing their differences?
I think this will also interfere if there in there an 'unknown' value of 361?

**Review by mickmis — APPROVED** [review]
LGTM modulo the modulo issue
(and the passing CI)

File: monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py (hunk 1 above)
Comment (the-glu) [reply monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py:484]: There was already that in previous function, I do assume -2 == 358 and that on purpose ?

There is indeed an issue with 361 however

File: monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py [hunk 2]
```diff
@@ -446,51 +442,56 @@ def value_comparator(v1: Optional[float], v2: Optional[float]) -> bool:
...
-        else:
-            self._test_scenario.record_note(
-                key="skip_reason",
-                message=f"Unsupported version {self._rid_version}: skipping track direction evaluation",
-            )
+        def value_validator(val: float) -> float:
+            if math.isclose(val, constants.SpecialTrackDirection):
+                return val
+
+            if val < constants.MinTrackDirection:
```
Comment (mickmis) [inline monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py:458]: `constants.MinTrackDirection` is actually wrong: the value is `-359` while the OpenAPI specifies `0` :/ 
We should update `uas_standards`.

**Review by mickmis — APPROVED** [review]

File: monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator_test.py [hunk 3]
```diff
@@ -1257,3 +1229,96 @@ def dp_field_setter(flight: Any, value: T) -> Any:
...
+
+    _assert_generic_evaluator_dont_have_default(
+        *base_args,
+        valid_value=42,
+        valid_value_2=3.14,
+    )
+
+    # Resolution is in steps of 1
+    # Float values are funny, we cannot test 0.25 because check may 'round' that to 0.04999999999999716
+    for v1 in [42, 3.14, 100, -100]:
```
Comment (mickmis) [inline monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator_test.py:1291]:
```suggestion
    for v1 in [42, 3.14, 100]:
```
